packages/main/src/FileUploader.ts

```diff
@@ -317,6 +360,33 @@ class FileUploader extends UI5Element implements IFormInputElement {
 		}, "");
 	}
 
+	/**
+	 * Checks whether all files are below `maxFileSize` (if set),
+	 * and fires a `fileSizeExceeded` event if any file exceeds it.
+	 *
+	 * @returns true if all files are below `maxFileSize`
+	 * @private
+	 */
+	_validateFileSize(files: FileList | null): boolean {
+		if (!this.maxFileSize || !files) {
+			return true;
+		}
+
+		const convertBytesToMegabytes = (bytes: number) => (bytes / 1024) / 1024;
+
+		for (let i = 0; i < files.length; i++) {
+			const fileSize = convertBytesToMegabytes(files[i].size);
+			if (fileSize > this.maxFileSize) {
+				this.fireEvent<FileUploaderFileSizeExceededEventDetail>("fileSizeExceeded", {
```

Comment:
The event should be fired only once per user attempt to upload, not for every file that exceeds the limit. Is there a specific requirement to fire it multiple times? It would be best to collect all files that exceed the size in the loop and then only fire the event once after the loop with info about all files that exceed. This would also require a change to the event data (make it an array of what it currently is)

Reply:
Yeah, my bad about the event being fired multiple times. I noticed the `return` later.

Regarding my proposal - the event will still fire if even one file exceeds the limit (and prevent the upload), however it will have info about all files that exceeded the limit, not just the first one.

Reply:
The event was changed to return a `filesData` array containing objects with the `fileName` and `fileSize` of each file, but this might need some more feedback.